examples/crm/src/contacts/ContactImportButton.tsx

```diff
@@ -0,0 +1,264 @@
+import UploadIcon from '@mui/icons-material/Upload';
+import { Box, CircularProgress, Stack } from '@mui/material';
+import Alert from '@mui/material/Alert';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import * as Papa from 'papaparse';
+import { useState } from 'react';
+import {
+    Button,
+    FileField,
+    FileInput,
+    Form,
+    useDataProvider,
+    useGetIdentity,
+    useNotify,
+    useRefresh,
+} from 'react-admin';
+import { Link } from 'react-router-dom';
+import * as sampleCsv from './contacts_export.csv?raw';
+import { ContentImportSchema, useContactImport } from './useContactImport';
+
+const SAMPLE_URL = `data:text/csv;charset=utf-8,${encodeURIComponent(sampleCsv.default)}`;
+
+export const ContactImportButton = () => {
+    const refresh = useRefresh();
+    const user = useGetIdentity();
+    const dataProvider = useDataProvider();
+    const notify = useNotify();
+    const contactImports = useContactImport();
+
+    const [file, setFile] = useState<File | null>(null);
+
+    const [importState, setImportState] = useState<
+        'hidden' | 'idle' | 'running' | 'error' | 'complete'
+    >('hidden');
+
+    const [errorCount, setErrorCount] = useState(0);
+    const [importCount, setImportCount] = useState(0);
+
+    const handleOpenModal = () => {
+        setImportState('idle');
+        setErrorCount(0);
+        setImportCount(0);
+    };
+
+    const handleCancel = () => {
+        setImportState('hidden');
+    };
+
+    const handleFileChange = (file: File | null) => {
+        setFile(file);
+    };
+
+    const startImport = async () => {
+        if (!file) {
+            return;
+        }
+
+        setImportState('running');
+
+        Papa.parse<ContentImportSchema>(file, {
```

Comment:
I'm just wondering if this part shouldn't be in a separate file.

Reply:
I think this part can be factorized when implementing company import if it's good for you ?